packages/json-schema/src/json-schema-emitter.ts

```diff
@@ -532,7 +532,7 @@ export class JsonSchemaEmitter extends TypeEmitter<Record<string, any>, JSONSche
     const toEmit: EmittedSourceFile[] = [];
 
     for (const sf of sourceFiles) {
-      const emittedSf = this.emitter.emitSourceFile(sf);
+      const emittedSf = await this.emitter.emitSourceFile(sf);
```

Comment:
This is unfortunate breaking change of having to make `sourceFile()` async

Reply:
moved that breaking change into its own PR https://github.com/microsoft/typespec/pull/2183